actions/AuthAction.js

```diff
@@ -142,44 +140,34 @@ async function checkForEarlyProductAccess(toggle, sdkKey, isInternal, org) {
   });
 }
 
-function asAuthAction(action) {
+function asAuthAction(action, isAuthorize = true) {
   return async (params) => {
     const imsEndpoint = params.IMS_ENDPOINT;
     const clientId = params.IMS_CLIENT_ID;
-    const serviceClientId = params.IMS_SERVICE_CLIENT_ID;
-    const clientSecret = params.IMS_SERVICE_CLIENT_SECRET;
-    const permAuthCode = params.IMS_SERVICE_PERM_AUTH_CODE;
     const productContext = params.IMS_PRODUCT_CONTEXT;
     const earlyAccessToggle = params.FT_EARLY_ACCESS;
     const ldSdkKey = params.LD_SDK_KEY;
 
-    const accessToken = getAccessToken(params);
     const imsOrg = getImsOrg(params);
+    const accessToken = await getAccessToken(params);
 
-    // Validate the access token
-    if (!await isValidToken(imsEndpoint, clientId, accessToken)) {
-      throw new Error('Access token is invalid');
-    }
-
-    // Check that the profile has access to the product
-    const imsProfile = await getImsProfile(imsEndpoint, clientId, accessToken);
-    if (!imsProfile) {
-      throw new Error('Failed to fetch profile');
-    }
+    if (isAuthorize) {
+      // Check that the profile has access to the product
+      const imsProfile = await getImsProfile(imsEndpoint, clientId, accessToken);
+      if (!imsProfile) {
+        throw new Error('Failed to fetch profile');
```

Comment:
You added a bunch of new error messages. Are we going to show them to the user through UI? if yes, we should add localization for these strings

Reply:
@irenelagno , it looks like all error messages in this file are unlocalized, I see there is a ticket on our board to address that, https://jira.corp.adobe.com/browse/SITES-21934

Reply:
@slitviachenko let's put comment that the new messages need to be localized as well

Reply:
Curious to understand the options that we have for localizing the text in runtime actions as we don't have the flexibility of React components.

Reply:
One solution may be to:
-  move the localizable text from Runtime actions to the appropriate localization file in the frontend.
- send a coded message like `error.profileFetchFailedMessage` from the Runtime actions for errors.
- figure out a way to call the `formatMessage` function correctly to display the correct localized text.

Reply:
CC: @audrey-kho , ^^^ as you are working on https://jira.corp.adobe.com/browse/SITES-21934 , please take a look at @askayastha22 's suggestion

Reply:
I would add to @askayastha22 's suggestion that we might return an error message and an error code. The message itself is for developers who can open Chrome tools for debugging, while the error code will be used for translation purposes on the frontend.
For example,
```
{
  "error": {
    "code": 10005,
    "message": "The access token is not valid"
  }
}
```